### PR TITLE
Add some features

### PR DESCRIPTION
**Pull request by joshpme**
Patch Symfony
Allow additional conference information such as isbn and issn, series, series number and other information
Allow presented papers / unpublished proceedings via import and correctly formatted references
Fix up bibtex formatting

Requires ```php bin/console doctrine:migrations:migrate```

### CODE REVIEW

File: src/Entity/Conference.php [hunk 1]
```diff
@@ -40,11 +40,50 @@ class Conference implements \JsonSerializable
     /**
      * This is the date of the conference, Eg. May 2018
      * @var string
-     * @Assert\Regex("/^((?!May. )(May|[A-Z]{1}[a-z]{2}\.)(\-(?!May. )(May|[A-Z]{1}[a-z]{2}\.))?) [0-9]{4}$/", message="Please the correct format the date held in the format MMM YYYY")
```
Comment (joshpme) [inline src/Entity/Conference.php:43]: UI issues with this regex in browser. Symfony letting us down :(

File: config/services.yaml [hunk 2]
```diff
@@ -4,42 +4,54 @@
...
-        tags:
-            - { name: 'doctrine.event_listener', event: postConnect}
+  Doctrine\DBAL\Event\Listeners\OracleSessionInit:
+    tags:
+      - { name: "doctrine.event_listener", event: postConnect }
 
-    App\EventSubscriber\ConferenceSubscriber:
-        tags:
-            - { name: 'doctrine.event_subscriber', connection: 'default', priority: 2 }
+  Fvozar\Twig\OrdinalNumberExtension:
```
Comment (joshpme) [inline config/services.yaml:39]: Only real change is to add this ordial twig library. rest are formatting changes

File: public/js/global.js [hunk 3]
```diff
@@ -128,4 +128,9 @@ $(document).ready(function(){
         this.style.height = 'auto';
         this.style.height = (this.scrollHeight) + 'px';
     });
+});
+
+$('input[type=file]').on('change',function(e){
+    var fileName = e.target.files[0].name;
+    $(this).closest(".custom-file").find('.custom-file-label').html(fileName);
```
Comment (joshpme) [inline public/js/global.js:135]: This fixes a bug in the symfony bootstrap styling which prevents showing the file name in form fields.

File: src/Service/FormService.php [hunk 4]
```diff
@@ -41,7 +41,7 @@ public function toggleForm() {
 
     public function getForm() {
         if (!$this->getSession()->has("form")) {
-            return "long";
+            return "short";
```
Comment (joshpme) [inline src/Service/FormService.php:44]: Change default format to "short"

File: src/Service/ImportService.php [hunk 5]
```diff
@@ -51,11 +51,15 @@ private function findReferences($filename) {
                 }
             }
             if (isset($data[5])) {
-                $customDoi = trim($data[5]);
-                $reference->setCustomDoi($customDoi);
+                $inProc = !in_array(trim($data[5]),["no", "3", "4", "5"]);
```
Comment (joshpme) [inline src/Service/ImportService.php]: Allows yes / no. but also works is backwards compatible with Volkers scripts.

File: templates/conference/edit.html.twig [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
+{% block javascripts %}
+    {{ parent() }}
+    <script src="{{ asset("js/jquery.mask.min.js") }}"></script>
+    <script>
+    $(document).ready(function(){
+    $(":input").inputmask();
+    });
+    </script>
+{% endblock %}
+
```
Comment (joshpme) [inline templates/conference/edit.html.twig:12]: Allow input masking for isbn and issn

**Review by joshpme — COMMENTED** [review]
Approve me please :)

**Review by anstowebmaster — APPROVED** [review]

**Comment by anstowebmaster** [comment]
These changes approved on the 03/03/2022 by AB

**Review by anstowebmaster — APPROVED** [review]

**Pull request merged**